src/main.rs

```diff
@@ -176,28 +170,26 @@ fn main() {
         let opt = opt.clone();
         thread::spawn(move || {
             let mut _start = Instant::now();
-            let mut v: Vec<Option<Hashrate>> = vec![None; opt.threads];
-            for hash in hash_recv {
-                if opt.hashrate > 0 {
-                    let duration: Duration = _start.elapsed();
-                    v[hash.worker_id as usize] = Some(hash.borrow().clone());
-                    if ! v.contains(&None) && duration.as_secs_f32() > 30.0 {
-                        let mut total: f32 = 0.0;
-                        for h in v.iter() {
-                            let h2 = h.as_ref().unwrap();
-                            if opt.hashrate > 1 {
-                                println!("{}", h2);
-                            }
-                            total += h2.value();
-                        }
-                        let (total, unit) = hashrate::get_unit(total);
-                        println!("{} = {} {} ({})",
-                                    "Total Hashrate".blue(),
-                                    format!("{:.3}", total).red(),
-                                    unit,
-                                    format!("{} Workers", v.len()).yellow());
-                        _start = Instant::now();
+            let mut v = vec![Hashrate::default(); opt.threads];
+            for (worker_id, hash) in hash_recv {
+                let duration: Duration = _start.elapsed();
+                v[worker_id as usize] = hash;
+                if duration.as_secs_f32() > 30.0 {
+                    for h in v.iter() {
+                        println!(
+                            "Worker {} Hashrate = {}",
+                            format!("#{}", worker_id).yellow(),
+                            h
+                        );
                     }
+                    let total_rate = v.iter().cloned().sum::<Hashrate>();
+                    println!(
+                        "{} = {} ({})",
+                        "Total Hashrate".blue(),
+                        total_rate,
```

Comment:
will this print total hash rate with unit?

Reply:
Yeah

Reply:
Very nice! solution :) Loved it!